contracts/0.4.24/nos/NodeOperatorsRegistry.sol

```diff
@@ -1047,136 +1252,108 @@ contract NodeOperatorsRegistry is AragonApp, IStakingModule, Versioned {
     /// @notice distributes rewards among node operators
     /// @return the amount of stETH shares distributed among node operators
     function _distributeRewards() internal returns (uint256 distributed) {
-        IStETH stETH = IStETH(STETH_POSITION.getStorageAddress());
+        IStETH stETH = IStETH(getLocator().lido());
 
         uint256 sharesToDistribute = stETH.sharesOf(address(this));
         if (sharesToDistribute == 0) {
             return;
         }
 
-        (address[] memory recipients, uint256[] memory shares) = getRewardsDistribution(sharesToDistribute);
+        (address[] memory recipients, uint256[] memory shares, bool[] memory penalized) =
+            getRewardsDistribution(sharesToDistribute);
 
         distributed = 0;
+
         for (uint256 idx = 0; idx < recipients.length; ++idx) {
             if (shares[idx] == 0) continue;
+            if (penalized[idx]) {
+                /// @dev half reward punishment
+                /// @dev ignore remainder since it accumulated on contract balance
+                shares[idx] >>= 1;
+                stETH.transferShares(getLocator().burner(), shares[idx]);
```

Comment:
```solidity
IBurner(getLocator().burner()).requestBurnShares(address(this), shares[idx]);
```
allowance needed (maybe ∞) from NOR to Burner

Reply:
done